### ISSUE TITLE
Inabox Host to deal with duplicate event registers correctly

### ISSUE THREAD

**Issue by zhouyx** [ampproject/amphtml#27899]
Inabox `iframeClient` (receiver side) allows one to register multiple handler to the same event type. 
https://github.com/ampproject/amphtml/blob/fcf2c2ff8760cc07963cfed7edf953894c78894c/3p/iframe-messaging-client.js#L86-L90

However the inabox host (sender side) doesn't handle duplicate event registration.
https://github.com/ampproject/amphtml/blob/fcf2c2ff8760cc07963cfed7edf953894c78894c/ads/inabox/inabox-messaging-host.js#L49-L54
Overriding the callback without unregister previous one will lead to the identical event entry been sent twice. 

This will soon block the `IntersectionObserverPolyfill` work. 

cc @dvoytenko @ampproject/wg-ads

**Comment by dvoytenko** [comment]
I assume the underlying issue is still open? This is just the preparation to the actual implementation, right?

**Comment by zhouyx** [comment]
The PR has fixed the underlying issue by not allowing duplicate event handlers to be registered on the inabox host side.

**Comment by zhouyx** [comment]
Turned out we are all good. This is a non issue, I've reverted the change in #27912.

There're two types of messages that the inabox host support. 
`FULL_OVERLAY_FRAME` and `CANCEL_FULL_OVERLAY_FRAME`. Where we *do* want to invoke the event handler whenever a message is received. 

With `SEND_POSITIONS`, inabox host only want to invoke the event handler once, because it doesn't want to add two listeners to the position change event. But that has already been protected by checking the `this.iframeMap_[request.sentinel].observeUnregisterFn`. 
https://github.com/ampproject/amphtml/blob/fdbd68810cbb88d0fdbc20ac1080e29429a19575/ads/inabox/inabox-messaging-host.js#L186-L190

In summary, no change is required. It's safe to call `IframeClient.makeRequest` multiple times to register multiple callbacks.
